File: 0328-odd-even-linked-list/0328-odd-even-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* oddEvenList(ListNode* head) {
        if(head==NULL || head->next==NULL)
        {
            return head;
        }
        ListNode * temp=head;
        vector<int>arr;
        while(temp!=NULL && temp->next!=NULL)
        {
            arr.push_back(temp->val);
            temp=temp->next->next;
        }
        if(temp)
            arr.push_back(temp->val);
        temp=head->next;
         while(temp!=NULL && temp->next!=NULL)
        {
            arr.push_back(temp->val);
            temp=temp->next->next;
        }
        if(temp)
            arr.push_back(temp->val);
        temp=head;
        int i=0;
        while(temp)
        {
            temp->val=arr[i];
            i++;
            temp=temp->next;
        }
        return head;
    }
};

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
class Solution {
    public ListNode oddEvenList(ListNode head) {
          if (head == null || head.next == null) {
            return head; // handle empty or single-node list
        }
        ListNode odd = head;
        ListNode even = head.next;
        ListNode evenHead = head.next;
        while(even != null && even.next != null){
            odd.next = odd.next.next;
            even.next = even.next.next;
            odd = odd.next;
            even = even.next;
        }
        odd.next = evenHead;
        return head;
    }
}